src/libvirt-xml-create.js

```diff
@@ -249,22 +249,27 @@ export function getSnapshotXML(name, description, disks, memoryPath, isExternal,
             // Disk can have attribute "snapshot" set to "no", which means no snapshot should be created of the said disk
             // This cannot be configured through cockpit, but we should uphold it nevertheless
             // see "snapshot" attribute of <disk> element at https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms
-            if (disk.snapshot !== "no") {
-                const diskElem = doc.createElement('disk');
-                diskElem.setAttribute('name', disk.target);
-                diskElem.setAttribute('snapshot', 'external');
-
-                if (disk.type === "volume") {
-                    const poolPath = getStoragePoolPath(storagePools, disk.source.pool, connectionName);
-                    if (poolPath) {
-                        const sourceElem = doc.createElement('source');
-                        sourceElem.setAttribute('file', `${poolPath}/${disk.source.volume}.snap`);
-                        diskElem.appendChild(sourceElem);
-                    }
+            if (disk.snapshot == "no")
+                return;
+
+            // Skip disks without source.
+            if (isExternal && disk.type == "file" && !disk.source.file)
```

Comment:
This is interesting -- so if you don't include a disk in a snapshot, and revert it, the disk will still be present? Apparently so, as our integration test checks that `sdd` is still present after reverting.

 When I saw that code the first time a few months ago, I was a bit afraid that the entire disk would get dropped from the VM on reverting. Do you know, would this also work for a disk *with* a source file? Like, you could revert your main/OS partition to an earlier state, but keep your separate /home or /opt disk?